marlowe-dashboard-client/src/Welcome/Types.purs

```diff
@@ -1,36 +1,24 @@
 module Welcome.Types
   ( State
-  , Card(..)
+  , Modal(..)
   , Action(..)
   , WalletNicknameOrIdError(..)
   ) where
 
 import Prologue
 import Analytics (class IsEvent, defaultEvent, toEvent)
 import Clipboard (Action) as Clipboard
+import Contacts.Types (WalletDetails, WalletLibrary, WalletNicknameError, WalletNickname)
+import Data.Tuple.Nested (type (/\))
 import InputField.Types (Action, State) as InputField
 import InputField.Types (class InputFieldError)
 import Marlowe.PAB (PlutusAppId)
 import Types (WebData)
-import Contacts.Types (WalletDetails, WalletLibrary, WalletNickname, WalletNicknameError)
 
--- TODO (possibly): The Contacts submodule used in the Dashboard has some properties and
--- functionality that's similar to some of what goes on here. It might be worth generalising it so
--- it works in both cases, and including it as a submodule here too.
 type State
-  = { card :: Maybe Card
-    -- Note [CardOpen]: As well as making the card a Maybe, we add an additional cardOpen flag.
-    -- When closing a card we set this to false instead of setting the card to Nothing, and that
-    -- way we can use CSS transitions to animate it on the way out as well as the way in. This is
-    -- preferable to using the Halogen.Animation module (used for toasts), because in this case we
-    -- need to simultaneously animate (fade in/out) the overlay, and because the animation for
-    -- cards has to be different for different screen sizes (on large screens some cards slide in
-    -- from the right) - and that's much easier to do with media queries.
-    , cardOpen :: Boolean
+  = { modal :: Maybe (Modal /\ Boolean)
```

Comment:
This is definitely better. But I wouldn't lose the comment - it would be easy to forget what the Boolean is for (or for someone else later to wonder what it's doing). The comment just needs rewording a bit to match the new structure.

Also, one day you might figure out a nicer way of doing it with the Halogen.Animation module (the tricky bit will be getting it to work simply on screens of all sizes, which I couldn't crack).

Reply:
I think this approach is fine - I generally prefer CSS animations where possible anyway. They are much more efficient